### PR TITLE
Add empty tags message

### LINKED ISSUE #558: Add hint text to navigation drawer
Currently, when no tags exist, the tags section is hidden in the navigation drawer. Let's add some hint text in place of it to indicate that tags will appear there.

- If no tags exist then hide "Edit" button
- If no tags exist then show hint text

Screenshot 2019-12-17 at 16 58 39

### PR DESCRIPTION
**Pull request by zacharynguyeenx**
### Fix

[//]: # (Required - Add a concise description of what you fixed.  If this is related to an issue, add a link to it.  If applicable, add screenshots, animations, or videos to help illustrate the fix.)

Add an empty state message to the Tags section header and show it when there are no tags. 
Check commit messages for list of change.

Closes https://github.com/Automattic/simplenote-ios/issues/558

| No tags | With tags |
| - | - |
| image | image |

### Note

Message is not localised yet.

### Test
[//]: # (Required - List the steps to test the behavior.  For example:)
1. Launch fresh app
1. Open sidebar
    - [ ] Verify empty message is visible
    - [ ] Verify Edit button is hidden
1. Add tags to posts
1. Open sidebar
    - [ ] Verify empty message is hidden
    - [ ] Verify Edit button is visible
1. Delete all tags
    - [ ] Verify empty message is visible
    - [ ] Verify Edit button is hidden

### Review
[//]: # (Required - Add instructions for reviewers.  For example:)

Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
[//]: # (Required - Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:)
[//]: # (If the changes should not be included in release notes, add a statement to this section. For example:)

These changes do not require release notes.

### CODE REVIEW

**Comment by SylvesterWilmott** [comment]
Looks good! As mentioned [here](https://github.com/Automattic/simplenote-ios/issues/558), let's hide the "Untagged Notes" filter while the tags list is empty.

Also would love to get editorial for a review of the copy. I'm not too keen on what we have so far for this.

File: Simplenote/Classes/SPTagHeaderView.swift [hunk 1]
```diff
@@ -19,6 +19,16 @@ class SPTagHeaderView: UIView {
...
+    /// Message label
+    @IBOutlet private(set) var messageLabel: UILabel! {
+        didSet {
+            messageLabel.text = NSLocalizedString(
+                "Once you add tags to your notes, your tags will appear here.",
+                comment: "Message displayed when no tags exist"
+            )
+        }
+    }
+
```
Comment (jleandroperez) [inline Simplenote/Classes/SPTagHeaderView.swift]: Potentially speaking, a specific UITableViewCell specifically implemented for this is a best fit.

**Why:** UITableView is known to have issues whenever the TableSectionHeaderView changes its height, by using a cell we'd just avoid the entire issue.

**Review by jleandroperez — CHANGES_REQUESTED** [review]
Hi there @zacharynguyeenx !!

Added a note, perhaps it's best to avoid reusing the TagHeaderView for this. And second thought is: we'd probably need to hide the **Untagged Notes** button whenever there are no tags, since it's expected to match the contents of **All Notes**.

Thank you for helping us improve Simplenote!!

File: Simplenote/Classes/SPTagHeaderView.swift (hunk 1 above)
Comment (zacharynguyeenx) [reply Simplenote/Classes/SPTagHeaderView.swift]: As per the design there is no separator between the section header and the message cell. However currently the table view has its `separatorStyle` set to `singleLine`. To mix and match separator style would require manually adding the separators to the cells which might complicate things for us. Do we want to go through that much to add just a message here?

**Comment by SylvesterWilmott** [comment]
Closing this issue as it's no longer relevant.

**Pull request closed without merging**